apps/namadillo/src/App/Transfer/SelectToken.tsx

```diff
@@ -0,0 +1,379 @@
+import { Chain } from "@chain-registry/types";
+import { Modal, Stack } from "@namada/components";
+import { ModalTransition } from "App/Common/ModalTransition";
+import { Search } from "App/Common/Search";
+import {
+  connectedWalletsAtom,
+  getAvailableChains,
+  getChainRegistryByChainName,
+  getNamadaChainRegistry,
+  namadaRegistryChainAssetsMapAtom,
+} from "atoms/integrations";
+import { tokenPricesFamily } from "atoms/prices/atoms";
+import clsx from "clsx";
+import { useWalletManager } from "hooks/useWalletManager";
+import { KeplrWalletManager } from "integrations/Keplr";
+import { useAtom, useAtomValue } from "jotai";
+import { useMemo, useState } from "react";
+import { IoClose } from "react-icons/io5";
+import { AssetWithAmountAndChain } from "types";
+import { AddressDropdown } from "./AddressDropdown";
+import { ChainBadge } from "./ChainBadge";
+import { isNamadaAddress } from "./common";
+
+type SelectTokenProps = {
+  setSourceAddress: (address: string) => void;
+  sourceAddress: string;
+  destinationAddress: string;
+  isOpen: boolean;
+  onClose: () => void;
+  onSelect:
+    | ((
+        selectedAsset: AssetWithAmountAndChain,
+        newSourceAddress?: string
+      ) => void)
+    | undefined;
+  keplrWalletManager?: KeplrWalletManager | undefined;
```

Comment:
You should decide if you want to use `?` or `| undefined`

Reply:
Roger, here you are: https://github.com/namada-net/namada-interface/pull/2031/commits/5b0ba1dd79c7eff1827d43c67678ae1a1a2611fd